frontend/src/api/k8s/__tests__/notebooks.spec.ts

```diff
@@ -0,0 +1,936 @@
+      model: NotebookModel,
+      queryOptions: { name: mockNotebook.metadata.name, ns: mockNotebook.metadata.namespace },
+      patches: [
+        startPatch,
+        getTolerationPatch(tolerationChanges),
+        getPipelineVolumePatch(),
+        getPipelineVolumeMountPatch(),
+      ],
+    });
+    expect(createElyraServiceAccountRoleBinding).toHaveBeenCalledWith(mockNotebook);
+    expect(k8sPatchResourceMock).toHaveBeenCalledTimes(1);
+    expect(renderResult).toStrictEqual(mockNotebook);
+  });
+  it('should start a notebook with pipelines and replace tolerations', async () => {
+    const tolerationChanges = {
+      type: 'replace',
+      settings: [] as Toleration[],
+    } as TolerationChanges;
+    const enablePipelines = true;
+    const mockNotebook = mockNotebookK8sResource({ uid });
+
+    k8sPatchResourceMock.mockResolvedValue(mockNotebookK8sResource({ uid }));
+
+    const renderResult = await startNotebook(mockNotebook, tolerationChanges, enablePipelines);
+
+    expect(k8sPatchResourceMock).toHaveBeenCalledWith({
+      model: NotebookModel,
+      queryOptions: { name: mockNotebook.metadata.name, ns: mockNotebook.metadata.namespace },
+      patches: [
+        startPatch,
+        getTolerationPatch(tolerationChanges),
+        getPipelineVolumePatch(),
+        getPipelineVolumeMountPatch(),
+      ],
+    });
+    expect(createElyraServiceAccountRoleBinding).toHaveBeenCalledWith(mockNotebook);
+    expect(k8sPatchResourceMock).toHaveBeenCalledTimes(1);
+    expect(renderResult).toStrictEqual(mockNotebook);
+  });
+  it('should start a notebook with pipelines and remove tolerations', async () => {
+    const tolerationChanges = {
+      type: 'remove',
+      settings: [] as Toleration[],
+    } as TolerationChanges;
+    const enablePipelines = true;
+    const mockNotebook = mockNotebookK8sResource({ uid });
+
+    k8sPatchResourceMock.mockResolvedValue(mockNotebookK8sResource({ uid }));
+
+    const renderResult = await startNotebook(mockNotebook, tolerationChanges, enablePipelines);
+
+    expect(k8sPatchResourceMock).toHaveBeenCalledWith({
+      model: NotebookModel,
+      queryOptions: { name: mockNotebook.metadata.name, ns: mockNotebook.metadata.namespace },
+      patches: [
+        startPatch,
+        getTolerationPatch(tolerationChanges),
+        getPipelineVolumePatch(),
+        getPipelineVolumeMountPatch(),
+      ],
+    });
+    expect(createElyraServiceAccountRoleBinding).toHaveBeenCalledWith(mockNotebook);
+    expect(k8sPatchResourceMock).toHaveBeenCalledTimes(1);
+    expect(renderResult).toStrictEqual(mockNotebook);
+  });
+  it('should start a notebook without pipelines', async () => {
+    const tolerationChanges = {
+      type: 'nothing',
+      settings: [] as Toleration[],
+    } as TolerationChanges;
+    const enablePipelines = false;
+
+    k8sPatchResourceMock.mockResolvedValue(mockNotebookK8sResource({ uid }));
+    const mockNotebook = mockNotebookK8sResource({ uid });
```

Comment:
If both mock instances are related, create a single instance and reuse it in the test function.